File: hw_15/F_new.cpp
#include <iostream>
using namespace std;

const int N = 2 * 1000000;
int prime_set[N];

void sieve() {
    prime_set[0] = prime_set[1] = 0;
    for (int i = 2; i < N; i++) {
        prime_set[i] = 1;
    }
    for (int i = 2; i * i < N; i++) {
        if (prime_set[i]) {
            for (int j = i * i; j < N; j += i) {
                prime_set[j] = 0;
            }
        }
    }
}

int main() {
    int m;
    cin >> m;
    sieve();
    while (m) {
        int n;
        cin >> n;
        if (prime_set[n]) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
        m--;
    }
    return 0;
}

File: hw_15/F_new.py
import sys

N = 2 * 1000000
prime_set = [0, 0] + [1] * N

for i in range(2, N):
    if prime_set[i]:
        for j in range(i * i, N, i):
            prime_set[j] = 0


m = int(input())

while m:
    n = int(sys.stdin.readline())

    if prime_set[n]:
        print('YES')
    else:
        print('NO')

    m -= 1